src/EmbedIO/Sessions/LocalSessionManager.cs

```diff
@@ -239,6 +239,18 @@ public void Delete(IHttpContext context, string id)
             context.Response.Cookies.Add(BuildSessionCookie(string.Empty));
         }
 
+        /// <inheritdoc />
+        public void Delete(IHttpContext context)
+        {
+            var id = GetSessionId(context);
+
+            lock (_sessions)
+            {
+                if (!string.IsNullOrEmpty(id) && _sessions.TryGetValue(id!, out _))
+                    Delete(context, id);
```

Comment:
If `Delete(context, id)` goes away, you can simplify here by just moving the code and eliminating the other `Delete` method.